Assignment_1/palindrome_permutation.js

```diff
@@ -0,0 +1,33 @@
+const palindromePermutation = (str) => {
+
+    const charMap = new Map(); //frequency of each char in string
+    str = str.replace(/\s/g, '').toLowerCase();
+    //remove spaces and convert string to lower case
+  //
+    for (const char of str) {
+        if (charMap.has(char)) {
+          charMap.set(char, charMap.get(char) + 1);
+        } else {
+          charMap.set(char, 1);
+        }
```

Comment:
Although there's nothing wrong with this, you can also take advantage of JS's nullish coalescing operator (`??`) and the fact that the `Map.prototype.get` method returns `undefined` if the key isn't found to do this in one line.
```suggestion
      charMap.set(char, (charMap.get(char) ?? 0) + 1);
```